guacamole/src/main/frontend/src/app/player/directives/textView.js

```diff
@@ -0,0 +1,172 @@
+         * that provided timestamp.
+         *
+         * @type {!Function}
+         */
+        seek: '&',
+
+        /**
+         * The current position within the recording.
+         *
+         * @type {!Number}
+         */
+        currentPosition: '='
+
+    };
+
+    config.controller = ['$scope', '$element', '$injector',
+            function guacPlayerController($scope, $element) {
+
+        /**
+         * The phrase to search within the text batches in order to produce the
+         * filtered list for display.
+         *
+         * @type {String}
+         */
+        $scope.searchPhrase = '';
+
+        /**
+         * The text batches that match the current search phrase, or all
+         * batches if no search phrase is set.
+         *
+         * @type {!TextBatch[]}
+         */
+        $scope.filteredBatches = $scope.textBatches;
+
+        /**
+         * Whether or not the key log viewer should be full-screen. False by
+         * default unless explicitly enabled by user interaction.
+         *
+         * @type {boolean}
+         */
+        $scope.fullscreenKeyLog = false;
+
+        /**
+         * Toggle whether the key log viewer should take up the whole screen.
+         */
+        $scope.toggleKeyLogFullscreen = function toggleKeyLogFullscreen() {
+            $element.toggleClass("fullscreen");
+        };
+
+        /**
+         * Filter the provided text batches using the provided search phrase to
+         * generate the list of filtered batches, or set to all provided
+         * batches if no search phrase is provided.
+         *
+         * @param {String} searchPhrase
+         *     The phrase to search the text batches for. If no phrase is
+         *     provided, the list of batches will not be filtered.
+         */
+        const applyFilter = searchPhrase => {
+
+            // If there's search phrase entered, search the text within the
+            // batches for it
+            if (searchPhrase)
+                $scope.filteredBatches = fuzzysort.go(
+                    searchPhrase, $scope.textBatches, {key: 'text'})
+                .map(result => result.obj);
+
+            // Otherwise, do not filter the batches
+            else
+                $scope.filteredBatches = $scope.textBatches;
+
+        };
+
+        // Reapply the filter to the updated text batches
+        $scope.$watch('textBatches', applyFilter);
+
+        // Reapply the filter whenever the search phrase is updated
+        $scope.$watch('searchPhrase', applyFilter);
+
+        /**
+         * Convert the given quantity of milliseconds into a human readable
+         * format, deferring to the time service for the implementation.
+         *
+         * @param {!number} value
+         *     The time to format, in milliseconds.
+         *
+         * @returns {!string}
+         *     A human readable string representing the provided number of
+         *     milliseconds.
+         */
```

Comment:
Same here - you can just annotate:

```js
/**
 * @borrows playerTimeService.formatTime
 */
```

and avoid multiple copies of the same docs.